gossip/infostore.go

```diff
@@ -113,32 +112,39 @@ func newInfoStore(nodeID proto.NodeID, nodeAddr util.UnresolvedAddr) *infoStore
 
 // newInfo allocates and returns a new info object using specified key,
 // value, and time-to-live.
-func (is *infoStore) newInfo(val interface{}, ttl time.Duration) *info {
+func (is *infoStore) newInfo(val []byte, ttl time.Duration) *Info {
 	is.seqGen++
 	now := monotonicUnixNano()
 	ttlStamp := now + int64(ttl)
 	if ttl == 0 {
 		ttlStamp = math.MaxInt64
 	}
-	return &info{
-		Val:       val,
-		Timestamp: now,
-		TTLStamp:  ttlStamp,
-		NodeID:    is.NodeID,
-		peerID:    is.NodeID,
-		seq:       is.seqGen,
+
+	value := proto.Value{
+		Bytes: val,
+		Timestamp: &proto.Timestamp{
+			WallTime: now,
+		},
+	}
+
+	return &Info{
+		Value:    value,
+		TTLStamp: ttlStamp,
+		NodeID:   is.NodeID,
+		PeerID:   is.NodeID,
+		Seq:      is.seqGen,
 	}
 }
 
-// getInfo returns an info object by key or nil if it doesn't exist.
-func (is *infoStore) getInfo(key string) *info {
+// getInfo looks up an Info object by its key.
+func (is *infoStore) getInfo(key string) *Info {
 	if info, ok := is.Infos[key]; ok {
 		// Check TTL and discard if too old.
 		if info.expired(time.Now().UnixNano()) {
 			delete(is.Infos, key)
-			return nil
+		} else {
```

Comment:
I actually prefer the original version of this function, it's clearer.  Why change it?